docs/guide/management/sdam-monitoring.txt

```diff
@@ -1,3 +1,438 @@
+      "$clusterTime": {
+        "clusterTime": "6746664774655803393",
+        "signature": {
+          "hash": "AAAAAAAAAAAAAAAAAAAAAAAAAAA=",
+          "keyId": 0
+        }
+      }
+    },
+    "newDescription": {
+      "address": "localhost:27017",
+      "error": null,
+      "roundTripTime": 0,
+      "lastUpdateTime": 1570830310333,
+      "lastWriteDate": "2019-10-11T21:45:02.000Z",
+      "opTime": {
+        "ts": "6746664774655803393",
+        "t": 16
+      },
+      "type": "RSPrimary",
+      "minWireVersion": 0,
+      "maxWireVersion": 7,
+      "maxBsonObjectSize": 16777216,
+      "maxMessageSizeBytes": 48000000,
+      "maxWriteBatchSize": 100000,
+      "me": "localhost:27017",
+      "hosts": [
+        "localhost:27017"
+      ],
+      "passives": [],
+      "arbiters": [],
+      "tags": [],
+      "setName": "rs",
+      "setVersion": 1,
+      "electionId": "7fffffff0000000000000010",
+      "primary": "localhost:27017",
+      "logicalSessionTimeoutMinutes": 30,
+      "$clusterTime": {
+        "clusterTime": "6746664774655803393",
+        "signature": {
+          "hash": "AAAAAAAAAAAAAAAAAAAAAAAAAAA=",
+          "keyId": 0
+        }
+      }
+    }
+  }
+
+The type can be one of the following values.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Type
+     - Description
+   * - Unknown
+     - Unknown server
+   * - Standalone
+     - Standalone server
+   * - Mongos
+     - Mongos proxy
+   * - PossiblePrimary
+     - Not checked yet, but another server thinks this is a primary
+   * - RSPrimary
+     - Primary server
+   * - RSSecondary
+     - Secondary server
+   * - RSArbiter
+     - Arbiter
+   * - RSOther
+     - See `the spec <https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#rsghost-and-rsother>`_ for more details
+   * - RSGhost
+     - See `the spec <https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#rsghost-and-rsother>`_ for more details
+
+serverHeartbeatStarted
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: js
+
+   { connectionId: 'localhost:27017' }
+
+serverHeartbeatSucceeded
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: js
+
+   { durationMS: 20,
+     reply: {
+       setName: "rs", setVersion: 1, electionId: new ObjectId(),
+       maxBsonObjectSize : 16777216, maxMessageSizeBytes : 48000000,
+       maxWriteBatchSize : 1000, localTime : new Date(),
+       maxWireVersion : 4, minWireVersion : 0, ok : 1,
+       hosts: ["localhost:32000", "localhost:32001"],
+       arbiters: ["localhost:32002"]
+     },
+     connectionId: 'localhost:27017' }
+
+serverHeartbeatFailed
+^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: js
+
+   { durationMS: 20,
+     failure: new MongoError('some error'),
+     connectionId: 'localhost:27017' }
+
+serverOpening
+^^^^^^^^^^^^^
+
+.. code-block:: js
+
+   { topologyId: 0, name: 'localhost:27017' }
```

Comment:
why not just put the named structure (the result of `console.dir`)

Reply:
I wound up using a hybrid approach, where I did the console.dir, but them simplified some fields.